ui/src/components/modal/RequestAliasModal.tsx

```diff
@@ -0,0 +1,30 @@
+import Modal from "./Modal"
+import React, {useCallback, useState} from "react";
+import {UserAddIcon} from "@heroicons/react/outline";
+
+type Props = { show: boolean, setShow: (show: boolean) => void, onOk: (alias: string) => void | Promise<void>}
+const RequestAliasModal: React.FC<Props> = ({show, setShow, onOk}) => {
+  const [ aliasValue, setAliasValue ] = useState<string>()
+
+  const setAliasForDID = useCallback(async () => aliasValue? onOk(aliasValue) : Promise.resolve(),
+   [aliasValue, onOk])
+
+  return (
+    <Modal title="Set Alias" description="Set your global alias" show={show} onOK={setAliasForDID} onClose={() => setShow(false)} renderIcon={() => (<UserAddIcon className="h-6 w-6 text-green-600" aria-hidden="true" />)}>
+      <label htmlFor="alias" className="sr-only">
```

Comment:
I feel in the future we may need more description here. Can I change that in the future? What is it. Why do I need it.

Reply:
True - I'll update in a later PR if that's ok, to avoid merge difficulties now I have a PR chain.

Reply:
Updated in https://github.com/dankelleher/solarium/pull/40